src/activities/discussions/DiscussionTopicEntity.js

```diff
@@ -37,6 +38,28 @@ export class DiscussionTopicEntity extends Entity {
 		return this._entity && this._entity.hasClass(Classes.discussions.hasPosts);
 	}
 
+	/**
+	 * @returns {number} Post rating scheme of discussion topic.
+	 */
+	postRatingSelection() {
+		const options = this.postRatingOptions();
+		const selected = options?.find(option => option?.selected);
+		return selected.value || NONE_RATING_TYPE;
+	}
+
+	/**
+	 * @returns {object[]} Post rating options of discussion topic.
+	 */
+	postRatingOptions() {
+		const entity = this._entity;
+		if (!entity) return;
+
+		const action = this._entity.getActionByName(Actions.discussions.topic.updateRatingType);
```

Comment:
Do you need an additional check to ensure the action exists? Normally it would be a function of the format `canUpdateRatingType`

Reply:
Yeah best to add one here. I'll put one in.